docs/docs/getting_started.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Getting Started
+tags: []
+---
+
+## A Quick Start
+
+The easiest way to use Noir is through `nargo` (inspired by Rust's `cargo`). You can use `noirup` to quickly install `nargo`:
+
+```bash
+curl -L noirup.dev | bash
+noirup
+```
+
+Since Noir is a backend-agnostic programming language for writing zero-knowledge proofs, we need to pair it with a proving backend.
+
+As an example, we can will use Aztec Lab's [Barretenberg backend](https://github.com/AztecProtocol/aztec-packages/tree/master/barretenberg). This is made easy by [`bbup`](https://github.com/AztecProtocol/aztec-packages/blob/master/barretenberg/bbup), which installs Barretenberg's [`bb` CLI tool](https://github.com/AztecProtocol/aztec-packages/tree/master/barretenberg):
+
+```bash
+curl -L bbup.dev | bash
+bbup
+```
+
+:::info
+
+_Always_ inspect what you run.
+
+- `noirup.dev` redirects to [this script](https://github.com/noir-lang/noirup/blob/main/install) and adds [this program](https://github.com/noir-lang/noirup/blob/main/noirup) to your path.
+- according to the [bbup documentation](https://github.com/AztecProtocol/aztec-packages/tree/master/barretenberg/bbup), `bbup.dev` redirects to [this script](https://github.com/AztecProtocol/aztec-packages/blob/master/barretenberg/bbup/install) which adds [this program](https://github.com/AztecProtocol/aztec-packages/blob/master/barretenberg/bbup/bbup.ts) globally, installing `node` if needed.
+
+:::
+
+## Nargo
+
+Nargo provides the ability to initiate and execute Noir projects. Let's initialize the traditional `hello_world`:
+
+```sh
+nargo new hello_world
+```
+
+Two files will be created.
+
+- `src/main.nr` contains a simple boilerplate circuit
+- `Nargo.toml` contains environmental options, such as name, author, dependencies, and others.
+
+Glancing at _main.nr_ , we can see that inputs in Noir are private by default, but can be labeled public using the keyword `pub`. This means that we will _assert_ that we know a value `x` which is different from `y` without revealing `x`:
+
+```rust
+fn main(x : Field, y : pub Field) {
+    assert(x != y);
+}
+```
+
+To learn more about private and public values, check the [Data Types](./noir/concepts/data_types/index.md) section.
+
+### Compiling and executing
+
+We can now use `nargo` to generate a _Prover.toml_ file, where our input values will be specified:
+
+```sh
+nargo check
+```
+
+Let's feed some valid values into this file:
+
+```toml
+x = "1"
+y = "2"
+```
+
+We're now ready to compile and execute our Noir program. By default the `nargo execute` command will do both, and generate the `witness` that we need to feed to our proving backend:
+
+```sh
+nargo execute hello-witness
```

Comment:
```suggestion
nargo execute
```

Following https://github.com/noir-lang/noir/pull/6031, `nargo execute` now writes a witness file named after the Noir package by default.